src/main/webapp/script.js

```diff
@@ -220,51 +220,117 @@ function getGraphDisplay(graphNodes, graphEdges) {
     node.tip.show();
   });
 
-  const searchElement = document.getElementById('search');
-  document.getElementById('search-button').onclick = function() {
-    if (searchNode(cy, searchElement.value) || searchElement.value == "") {
-      document.getElementById('search-error').innerText = "";
+  document.getElementById('reset').onclick = function(){ resetElements(cy) };
+
+  document.getElementById('search-button').onclick = function() { search(cy, "node", searchNode) };
+
+  document.getElementById('search-token-button').onclick = function() { search(cy, "token", searchToken) };
+}
+
+/**
+ * Searches based on type (node or token)
```

Comment:
Also document the return value and that this changes the highlighting of the found node.